middlewares/selectAndAuthzInstanceQuestion.sql

```diff
@@ -155,5 +155,6 @@ WHERE
     $assessment_id::BIGINT IS NULL
     OR a.id = $assessment_id
   )
+  AND q.deleted_at IS NULL
```

Comment:
Do we also want checks on `aq.deleted_at` and `a.deleted_at` here? Conceivably also `c.deleted_at`? I'm not sure how high up the chain it's worth enforcing?

Reply:
The first two make sense! I think `c.deleted_at` will (should) always be caught by `middlewares/authzCourseOrInstance`?

Reply:
I added checks for `aq.deleted_at` and `a.deleted_at` in b0c6979.

Reply:
Good point on course access being checked already.

Reply:
Is there a problem here with using `aq.deleted_at`? What happens if a student opens a question that was active when they took the assessment, but deleted afterwards?

Reply:
We check for `aq.deleted_at` in many places already (grep for `aq.deleted_at IS NULL`), so this is just making it consistent with everywhere else. If an assessment question is deleted, I think the correct thing to do is to not show it to students and forbid access to it even if they manage to guess the URL. Does that sound right to you?

Reply:
That makes sense. I didn't check the list, I was just confirming that this is the intended behaviour, particularly for exams, since they don't regenerate instance questions when the assessment changes.

Reply:
This has been a sore point for a long time, actually. What happens is that we are running an exam in the CBTF, the instructor realizes on day 2 that one question in an `alternative_group` is partially broken, and they want to remove this question from the assessment so that future students won't be assigned it. However, they want existing exams taken on day 1 to still show the question that the student did, especially because the question might have worked fine for these students (and we want to show how their score was computed).

I found the old issue about this: https://github.com/PrairieLearn/PrairieLearn/issues/1356

In summary, I think we want to allow access to instance questions even though the assessment question has been deleted. However, handling this cleanly will probably require some more code changes beyond just changing `deleted_at` filters, because we'll need to track the `alternative_group` and `zone` information that might have been deleted.

Reply:
Yes, let's _not_ add `aq.deleted_at IS NULL` here.

Reply:
Removed that check in fd5f67e.

Reply:
Whoops, there was one more that I removed in 2815b39.